### PR TITLE
Move Dashboard to a new module

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

Clean up site.js from the big Dashboard global definition it has.

Mainly intended to make the future site.js ES6/Webpack PR a bit smaller.

**Issues**

Part of #868

### CODE REVIEW

File: src/scripts/clientUtils.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
+export function getCurrentUser() {
+    return window.ApiClient.getCurrentUser(false);
+}
+
+//TODO: investigate url prefix support for serverAddress function
+export function serverAddress() {
+    if (AppInfo.isNativeApp) {
+        var apiClient = window.ApiClient;
```
Comment (matjaz321) [inline src/scripts/clientUtils.js:8]: const

File: src/scripts/clientUtils.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    }
+
+    var queryString = getWindowLocationSearch();
+
+    if (preserveQueryString && queryString) {
+        url += queryString;
+    }
+
+    return new Promise(function (resolve, reject) {
+        require(['appRouter'], function (appRouter) {
```
Comment (matjaz321) [inline src/scripts/clientUtils.js:90]: import

File: src/scripts/clientUtils.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    if (typeof options == 'string') {
+        return void import('toast').then(({default: toast}) => {
+            toast.default({
+                text: options
+            });
+        });
+    }
+
+    import('alert').then(({default: alert}) => {
+        alert.default({
```
Comment (Camc314) [inline src/scripts/clientUtils.js:146]:
```suggestion
        alert({

File: src/scripts/clientUtils.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+export function processServerConfigurationUpdateResult(result) {
+    Promise.all([
+        import('loading'),
+        import('toast')
+    ])
+        .then(([{default: loading}, {default: toast}]) => {
+            loading.hide();
+            toast.default(Globalize.translate('MessageSettingsSaved'));
```
Comment (Camc314) [inline src/scripts/clientUtils.js:115]:
```suggestion
            toast.(Globalize.translate('MessageSettingsSaved'));

File: src/scripts/clientUtils.js [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    alert({
+        title: status,
+        message: response.headers ? response.headers.get('X-Application-Error-Code') : null
+    });
+}
+
+export function alert(options) {
+    if (typeof options == 'string') {
+        return void import('toast').then(({default: toast}) => {
+            toast.default({
```
Comment (Camc314) [inline src/scripts/clientUtils.js:139]:
```suggestion
            toast({

File: src/scripts/clientUtils.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+export function processServerConfigurationUpdateResult(result) {
+    Promise.all([
+        import('loading'),
+        import('toast')
+    ])
+        .then(([{default: loading}, {default: toast}]) => {
+            loading.hide();
+            toast.(Globalize.translate('MessageSettingsSaved'));
```
Comment (Camc314) [inline src/scripts/clientUtils.js:115]:
```suggestion
            toast(Globalize.translate('MessageSettingsSaved'));
```

My Mistake

File: src/scripts/clientUtils.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+export function processPluginConfigurationUpdateResult() {
+    Promise.all([
+        import('loading'),
+        import('toast')
+    ])
+        .then(([{default: loading}, {default: toast}]) => {
+            loading.hide();
+            toast.default(Globalize.translate('MessageSettingsSaved'));
```
Comment (Camc314) [inline src/scripts/clientUtils.js:104]:
```suggestion
            toast(Globalize.translate('MessageSettingsSaved'));

File: src/scripts/clientUtils.js [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+export function capabilities(appHost) {
+    let capabilities = {
+        PlayableMediaTypes: ['Audio', 'Video'],
+        SupportedCommands: ['MoveUp', 'MoveDown', 'MoveLeft', 'MoveRight', 'PageUp', 'PageDown', 'PreviousLetter', 'NextLetter', 'ToggleOsd', 'ToggleContextMenu', 'Select', 'Back', 'SendKey', 'SendString', 'GoHome', 'GoToSettings', 'VolumeUp', 'VolumeDown', 'Mute', 'Unmute', 'ToggleMute', 'SetVolume', 'SetAudioStreamIndex', 'SetSubtitleStreamIndex', 'DisplayContent', 'GoToSearch', 'DisplayMessage', 'SetRepeatMode', 'SetShuffleQueue', 'ChannelUp', 'ChannelDown', 'PlayMediaSource', 'PlayTrailers'],
+        SupportsPersistentIdentifier: self.appMode === 'cordova' || self.appMode === 'android',
+        SupportsMediaControl: true
+    };
+    appHost.getPushTokenInfo();
+    return capabilities = Object.assign(capabilities, appHost.getPushTokenInfo());
```
Comment (dmitrylyzo) [inline src/scripts/clientUtils.js:161]: Untested
```suggestion
    return Object.assign(capabilities, appHost.getPushTokenInfo());
```

File: src/scripts/clientUtils.js [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    apiClient = apiClient || window.ApiClient;
+    window.ApiClient = apiClient;
+}
+
+export function logout() {
+    ConnectionManager.logout().then(function () {
+        let loginPage;
+
+        if (AppInfo.isNativeApp) {
+            loginPage = 'selectserver.html';
```
Comment (dmitrylyzo) [inline src/scripts/clientUtils.js:55]: Not sure, but we probably need to call `selectServer` instead.

Comment (MrTimscampi) [reply src/scripts/clientUtils.js:55]: I just copied stuff out of site.js, it's not meant to fix or refactor thing, only to get code out of my way for the Webpack changes :)

But yeah, we should change this afterwards.

**Pull request merged**